src/fixing/accumulated_edits.rs

```diff
@@ -0,0 +1,1473 @@
+use std::{cmp, iter::Peekable, ops};
+
+use itertools::{Either, Itertools};
+use squalid::{EverythingExt, OptionExt, VecExt};
+use tree_sitter_grep::{
+    ropey::Rope,
+    tree_sitter::{InputEdit, Point, Range},
+    RopeOrSlice,
+};
+
+#[derive(Clone, Debug)]
+pub struct AccumulatedEdits {
```

Comment:
This is pretty hairy but seemed like a useful abstraction (don't remember exactly where it's getting used)

I think it's probably still buggy and that that's related to the instability of the LSP server